### DISCUSSION TITLE
Best way to change the query parameter amp=1

### DISCUSSION THREAD

**Discussion by joafeldmann** [vercel/next.js#13402] [Help] [answered]
We're planning to migrate an existing site to next and therefore we're looking for a way to map/reuse our old AMP-query parameter `?output=amp` for all routes.

Is it somehow possible to modify the next amp query param? Or would a custom rewrite be an option? Something like

```
module.exports = {
  experimental: {
    async rewrites() {
      return [
        // rewrite output=amp to amp=1     
      ];
    },
  },
};
```

**Comment by Timer** [comment, marked as answer]
This currently isn't supported. You could open a feature request and see if it gains traction!

**Reply by joafeldmann** [reply]
Ok, thanks! Is there any way to solve this with a general rewrite rule?

**Reply by Timer** [reply]
There's not, unfortunately. Rewrites cannot consider input query strings (only can append).

**Reply by joafeldmann** [reply]
Ok, I am considering creating a feature request, but I would prefer a way to handle this by redirects/rewrites instead of customizing the amp param in next. This could be easily done with a reverse-proxy but when hosting next on vercel there is - as far as I know - no way to do this. 

Do you see a chance that custom routes / rewrites (#9081) might support renaming of query params at all?
